src/Components/Components/src/EventHandlerInvoker.cs

```diff
@@ -11,24 +11,30 @@ namespace Microsoft.AspNetCore.Components
     /// </summary>
     public readonly struct EventHandlerInvoker
     {
-        private readonly MulticastDelegate _delegate;
+        public static readonly EventHandlerInvokerFactory Factory = new EventHandlerInvokerFactory();
```

Comment:
@SteveSandersonMS  - what would you think about making this a more user-friendly name and making it part of the programming experience. I'm thinking `Command` and `Command<T>` and keeping the ability to abstract over the async-vs-non.

The reason why you might want to have something like this is to deal with the problem when a component wants to use delegates as parameters. The consumer of the component has to provide delegates with the signature that the component specifies. This is a little clunky in comparison to how event-handlers work which accept a variety of signatures seamlessly. This would be done in such a way that the user of a component doesn't need to know about the details of command, a super-set of what we have today would work. 

The other detail is that if a component accepts a `Command` as a parameter, it allows us to back in the same eventing behaviour that we have with event handlers and bind but without guesswork and conventions.

Reply:
This sounds superb. I was just hitting cases today with the forthcoming `EditForm` component where it's awkward to have an `OnSubmit` parameter because sometimes the consumer might legitimately want to supply any of `Action`, `Action<EditContext>`, `Func<Task>` or `Func<EditContext, Task>`.

If this could be declared by `EditForm` as `[Parameter] Command<EditContext> OnSubmit { get; set; }` and this makes it possible for the consumer to use any of those delegate types, that would be a huge win.

On naming, the name `Command` might be a good fit, or `EventCallback` is also pretty explict since it comes with the implication that it triggers post-event handling logic.

Reply:
If we want this to support arbitrary T it will require some doing, but shouldn't be that bad. The main that comes up is that it kinda has to be reference type at that point. This might be OK because we can amortize some of the allocations we already have to do - I'm going to have a go at this and we can look at in more detail. 

I want to spend some extra time to really nail this part since I think it nice to have a primitive that we use to events, bind, and arbitrary app functionality like `OnSubmit`.